client/api/index.js

```diff
@@ -75,3 +83,101 @@ export const createOrder = ( sourceEvent, paymentRequestType ) => {
 		url: getAjaxUrl( 'create_order' ),
 	} );
 };
+
+/**
+ * Handles generic connections to the server and Stripe.
+ */
+export default class WCStripeAPI {
+	/**
+	 * Prepares the API.
+	 *
+	 * @param {Object}   options Options for the initialization.
+	 * @param {Function} request A function to use for AJAX requests.
+	 */
+	constructor( options, request ) {
+		this.stripe = null;
+		this.options = options;
+		this.request = request;
+	}
+
+	/**
+	 * Generates a new instance of Stripe.
+	 *
+	 * @return {Object} The Stripe Object.
+	 */
+	getStripe() {
+		const { key, locale, isUPEEnabled } = this.options;
+
+		if ( ! this.stripe ) {
+			if ( isUPEEnabled ) {
+				this.stripe = new Stripe( key, {
+					betas: [ 'payment_element_beta_1' ],
+					locale,
+				} );
+			} else {
+				this.stripe = new Stripe( key, {
+					locale,
+				} );
+			}
+		}
+		return this.stripe;
+	}
+
+	/**
+	 * Load Stripe for payment request button.
+	 *
+	 * @return {Promise} Promise with the Stripe object or an error.
+	 */
+	loadStripe() {
+		return new Promise( ( resolve ) => {
+			try {
+				resolve( this.getStripe() );
+			} catch ( error ) {
+				// In order to avoid showing console error publicly to users,
+				// we resolve instead of rejecting when there is an error.
+				resolve( { error } );
+			}
+		} );
+	}
+
+	initSetupIntent() {
+		console.error( 'TODO: Not implemented yet: initSetupIntent' );
+	}
+
+	/**
+	 * Creates an intent based on a payment method.
+	 *
+	 * @param {number} orderId The id of the order if creating the intent on Order Pay page.
+	 *
+	 * @return {Promise} The final promise for the request to the server.
+	 */
+	createIntent( orderId ) {
+		return this.request( getAjaxUrl( 'create_payment_intent' ), {
+			stripe_order_id: orderId,
+			_ajax_nonce: getStripeServerData()?.createPaymentIntentNonce,
+		} )
+			.then( ( response ) => {
+				if ( ! response.success ) {
+					throw response.data.error;
+				}
+				return response.data;
+			} )
+			.catch( ( error ) => {
+				if ( error.message ) {
+					throw error;
+				} else {
+					// Covers the case of error on the Ajax request.
+					throw new Error( error.statusText );
+				}
+			} );
+	}
+
+	confirmIntent( url, savePaymentMethod ) {
+		console.error( 'TODO: Not implemented yet: confirmIntent' );
+		return true;
+	}
+
+	saveUPEAppearance( appearance ) {
+		console.error( 'TODO: Not implemented yet: saveUPEAppearance' );
+	}
```

Comment:
Are we implementing this now or in a different PR? 🤔

Reply:
We want to get this PR merged and continue UPE dev in smaller, more discreet PR's against develop now that we have the feature flag.